stdlib/task/src/main/ballerina/task/listener.bal

```diff
@@ -33,8 +33,9 @@ public type Listener object {
         }
     }
 
-    public function __attach(service s, map<any> annotationData) returns error? {
-        return self.register(s, annotationData);
+    public function __attach(service s, string? name = ()) returns error? {
+        // ignore param 'name'
+        return self.register(s, {});
```

Comment:
Is using `{}` this intentional? I think parsing service name to register function is useful. @ThisaruGuruge @rdhananjaya wdyt ?

Reply:
We need an `attachment` parameter to pass inside the service. We used this to pass that `attachment` parameter, if the developer provide it. We use it in [Task Scheduler](https://github.com/ballerina-platform/ballerina-lang/blob/746d2424ecd17f83b6d9b324c6203016a3d13967/stdlib/task/src/main/ballerina/task/scheduler.bal#L33)

Reply:
Shall we create an issue to track this, Ideally, with the new "Annotation and reflection API" design we should be able to handle this inside the method, without user providing this.

Reply:
Sure. I'll create one.